### ISSUE TITLE
SendRequest.tx is modified even when Wallet.completeTx() throws ExceededMaxTransactionSize exception

### ISSUE THREAD

**Issue by oscarguindzberg** [bitcoinj/bitcoinj#1359]
Scenario
- I call Wallet.completeTx(SendRequest)
- Tx generated is too big, so ExceededMaxTransactionSize exception is thrown

Result
- When Wallet.completeTx(SendRequest) finishes, the SendRequest's tx is modified: some inputs were added to it

Expected Result
- When Wallet.completeTx(SendRequest) finishes, the SendRequest's tx is unlatered (Same thing that happens when any other CompletionException subclass is thrown)

**Comment by schildbach** [comment]
Yes, this is kind of expected unfortunately. SendRequest is used like a register API, some are write only, some read only, and some both. I don't like it either, but we can improve on it I guess.
